src/Relation/Binary/Construction/Free/Supremum/Order.agda

```diff
@@ -0,0 +1,89 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Consequences on orders of freely adding an infimum to a Set
+------------------------------------------------------------------------
+open import Relation.Binary
+
+module Relation.Binary.Construction.Free.Supremum.Order
+       {a r} {A : Set a} (_≤_ : Rel A r) where
+
+open import Level
+open import Data.Sum as Sum
+open import Function.Equivalence using (equivalence)
+open import Relation.Nullary
+import Relation.Nullary.Decidable as Dec
+open import Relation.Binary.Construction.Free.Supremum
+
+data _≤⁺_ : Rel (A ⁺) r where
+  []≤[] : {k l : A} → k ≤ l → [ k ] ≤⁺ [ l ]
```

Comment:
Would this constructor be more readable as `[_]`? You'd then get `[ k≤l ]` which is more readable and intuitive than `[]≤[] k≤l` in my opinion.